packages/flutter/test/material/dropdown_test.dart

```diff
@@ -418,14 +422,16 @@ void main() {
 
     expect(value, equals('three'));
 
+    // Once the value of DropdownButton change, need rebuilding
+    // to update RenderIndexStack.
+    await tester.pumpWidget(build());
```

Comment:
The dropdown menu button need to rebuild automatically. If we have to manually pump a new widget, that is a bug in the dropdown menu button itself.

In other word, if a `tester.pumpAndSettle()` does not work, we need to fix the code in dropdown.dart instead.

Reply:
> The dropdown menu button need to rebuild automatically. If we have to manually pump a new widget, that is a bug in the dropdown menu button itself.
> 
> In other word, if a `tester.pumpAndSettle()` does not work, we need to fix the code in dropdown.dart instead.

 #35736 discusses why the widget does not automatically update the UI